src/components/TableProperty/TablePropertyAdmin.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { useEffect, useState } from 'react'
+import style from './TableProperty.module.scss'
+import {
+  Button,
+  Select,
+  TextInput,
+  Table,
+  Image,
+  Modal,
+  Pagination,
+} from '@mantine/core'
+import { FaSearch, FaEdit } from 'react-icons/fa'
+import { useDisclosure } from '@mantine/hooks'
+import ModalProperty from '../ModalProperty/ModalProperty'
+import { Category, Feature, Properties } from '@/types'
+import { axiosInstance } from '../../service/AxiosInstance'
+import Swal from 'sweetalert2'
+import { formatMoneyToUSD } from '../../utils/commonFunctions'
+import { useAppDispatch, useAppSelector } from '../../redux/hooks'
+import { getAllCategories } from '../../redux/reducers/categorySlice'
+import { getAllFeatures } from '../../redux/reducers/featureSlice'
+import { PiArrowsDownUp } from 'react-icons/pi'
+import {
+  searchPropertyForAdmin,
+  SearchProps,
+} from '../../service/SearchService'
+import {
+  CODE_RESPONSE_400,
+  CODE_RESPONSE_401,
+  CODE_RESPONSE_404,
+} from '../../constants/codeResponse'
+import { getAllPropertiesForAdminSerivce } from '../../service/PropertyService'
+import { MdDelete } from 'react-icons/md'
+
+const TablePropertyAdmin = () => {
+  const [opened, { open, close }] = useDisclosure(false)
+  const [selectedProperty, setSelectedProperty] = useState<Properties | null>(
+    null,
+  )
+  const [isUpdated, setIsUpdated] = useState(false)
+  const [properties, setProperties] = useState<Properties[]>([])
+  const [sort, setSort] = useState(true)
+  const [statusProperty, setStatusProperty] = useState(false)
+  const handlePropertyView = (property: Properties) => {
+    setSelectedProperty(property)
+    open()
+  }
+
+
+  const getAllProperties = async () => {
+    const res = await getAllPropertiesForAdminSerivce()
+    setProperties(res.metaData.data)
+    setTotalPages(res.metaData.totalPages)
+    setTotalItems(res.metaData.totalItems)
+  }
+
+  useEffect(() => {
+    getAllProperties()
+    setIsUpdated(false)
+  }, [isUpdated])
+  //Apply API getAllCategory
+  const dispatch = useAppDispatch()
+  const categories: Category[] = useAppSelector(
+    (state) => state.category.categoriesList,
+  )
+  useEffect(() => {
+    const promise = dispatch(getAllCategories())
+    return () => {
+      promise.abort()
```

Comment:
why are you using abort here?

Reply:
When you initialize this project, the Index file uses "StrictMode". So I use it to handle API to be called 1 time. But Ms.Hanh removed Strict so now, we don't need it. I'll remove it now.